app/src/main/java/com/keylesspalace/tusky/util/SaveTootHelper.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+public class SaveTootHelper {
+
+    public static final String TAG = "SaveTootHelper";
+
+    private TootDao tootDao;
+    private Context context;
+
+    public SaveTootHelper(@NonNull TootDao tootDao, @NonNull Context context) {
+        this.tootDao = tootDao;
+        this.context = context;
+    }
+
+
+    @SuppressLint("StaticFieldLeak")
+    public boolean saveToot(@NonNull String content,
+                             @NonNull String contentWarning,
+                             @Nullable String savedJsonUrls,
+                             @NonNull List<ComposeActivity.QueuedMedia> mediaQueued,
+                             int savedTootUid,
+                             @Nullable String inReplyToId,
+                             @Nullable String replyingStatusContent,
+                             @Nullable String replyingStatusAuthorUsername,
+                             @NonNull Status.Visibility statusVisibility) {
+
+        if (TextUtils.isEmpty(content) && mediaQueued.isEmpty()) {
+            return false;
+        }
+
+        // Get any existing file's URIs.
+        ArrayList<String> existingUris = null;
+        if (!TextUtils.isEmpty(savedJsonUrls)) {
+            existingUris = new Gson().fromJson(savedJsonUrls,
+                    new TypeToken<ArrayList<String>>() {
+                    }.getType());
+        }
+
+        String mediaUrlsSerialized = null;
+        if (!ListUtils.isEmpty(mediaQueued)) {
+            List<String> savedList = saveMedia(mediaQueued, existingUris);
+            if (!ListUtils.isEmpty(savedList)) {
+                mediaUrlsSerialized = new Gson().toJson(savedList);
+                if (!ListUtils.isEmpty(existingUris)) {
+                    deleteMedia(setDifference(existingUris, savedList));
+                }
+            } else {
+                return false;
+            }
+        } else if (!ListUtils.isEmpty(existingUris)) {
+            /* If there were URIs in the previous draft, but they've now been removed, those files
+             * can be deleted. */
+            deleteMedia(existingUris);
+        }
+        final TootEntity toot = new TootEntity(savedTootUid, content, mediaUrlsSerialized, contentWarning,
+                inReplyToId,
+                replyingStatusContent,
+                replyingStatusAuthorUsername,
+                statusVisibility);
+
+        new AsyncTask<Void, Void, Void>() {
+            @Override
+            protected Void doInBackground(Void... params) {
+                tootDao.insertOrReplace(toot);
+                return null;
+            }
+        }.execute();
+        return true;
+    }
+
+    @Nullable
+    private List<String> saveMedia(@NonNull List<ComposeActivity.QueuedMedia> mediaQueued,
+                                   @Nullable List<String> existingUris) {
+
+        File imageDirectory = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
```

Comment:
Does user expect that we're going to save their pictures from the toot to the public directory where any app can access them?..

Reply:
Thats no new code, its cut from `ComposeActivity`. And yes, its  probably a bad idea, I should change it.